js/plugins/vertexai/src/vector-search/indexers.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { embedMany } from '@genkit-ai/ai/embedder';
+import {
+  defineIndexer,
+  IndexerAction,
+  indexerRef,
+} from '@genkit-ai/ai/retriever';
+import z from 'zod';
+import {
+  Datapoint,
+  VertexAIVectorIndexerOptionsSchema,
+  VertexVectorSearchOptions,
+} from './types';
+import { upsertDatapoints } from './upsert_datapoints';
+
+/**
+ * Creates a reference to a Vertex AI indexer.
+ *
+ * @param {Object} params - The parameters for the indexer reference.
+ * @param {string} params.indexId - The ID of the Vertex AI index.
+ * @param {string} [params.displayName] - An optional display name for the indexer.
+ * @returns {Object} - The indexer reference object.
+ */
+export const vertexAiIndexerRef = (params: {
+  indexId: string;
+  displayName?: string;
+}) => {
+  return indexerRef({
+    name: `vertexai/${params.indexId}`,
+    info: {
+      label: params.displayName ?? `vertexAi - ${params.indexId}`,
+    },
+    configSchema: VertexAIVectorIndexerOptionsSchema.optional(),
```

Comment:
Why is this schema optional here but not in `defineIndexer` below?

Same for `retriever.ts`